__tests__/integration.test.js

```diff
@@ -1,30 +1,102 @@
-        return request(app).get("/api/topics").then((res) => {
-            expect(Array.isArray(res.body.topics)).toBe(true)
-            expect(res.body.topics).toHaveLength(3)
-            res.body.topics.forEach((topic, index) => {
-                expect(typeof res.body.topics[index]).toBe('object')
-                expect(topic).hasOwnProperty('slug')
-                expect(topic).hasOwnProperty('description')
-            });
-        })
-    });
-});
\ No newline at end of file
+  return seed(testData);
+});
+
+afterAll(() => {
+  return db.end();
+});
+
+describe("getTopics", () => {
+  it("should return a 200 status code", () => {
+    return request(app).get("/api/topics").expect(200);
+  });
+  it("should return an array of topic objects, each of which should have the following properties: slug, description) ", () => {
+    return request(app)
+      .get("/api/topics")
+      .then((res) => {
+        expect(res.body.topics).toHaveLength(3);
+        res.body.topics.forEach((topic, index) => {
+          expect(typeof res.body.topics[index]).toBe("object");
+          expect(topic).hasOwnProperty("slug");
+          expect(topic).hasOwnProperty("description");
+        });
+      });
+  });
+});
+
+describe("getAPI", () => {
+  it("should return an object describing all the available endpoints on your API", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((res) => {
+        const testObject = endpoints;
+        const endpointResult = res.body;
+        expect(testObject).toEqual(endpointResult);
+      });
+  });
+});
+
+describe("getArticleByID", () => {
+  it("should return correct article", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then((res) => {
+        const testArticle = res.body.article[0];
+        const expectedArticle = {
+          author: "butter_bridge",
+          title: "Living in the shadow of a great man",
+          article_id: 1,
+          body: "I find this existence challenging",
+          topic: "mitch",
+          created_at: "2020-07-09T20:11:00.000Z",
+          votes: 100,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        };
+        expect(testArticle).toEqual(expectedArticle);
+      });
+  });
+  it("should return a 404 error if no such id exists ", () => {
+    return request(app)
+      .get("/api/articles/999")
+      .expect(404)
+      .then((res) => {
+        expect(res.body.error).toBe("Article id 999 not found");
+      });
+  });
+  it("should return a 400 error if invalid ID passed ", () => {
+    return request(app)
+      .get("/api/articles/dog")
+      .expect(400)
+      .then((res) => {
+        expect(res.body.error).toBe("Bad request, dog is not a valid request");
```

Comment:
There is no need to have specific error messages, just a generic bad request message is enough